Purchases/Public/RCPurchases.m

```diff
@@ -503,17 +509,17 @@ - (void)productsWithIdentifiers:(NSArray<NSString *> *)productIdentifiers
     }
 
     if (missingProductIdentifiers.count > 0) {
-        [self.requestFetcher fetchProducts:missingProductIdentifiers
-                                completion:^(NSArray<SKProduct *> * _Nonnull newProducts) {
-                                    @synchronized (self) {
-                                        for (SKProduct *p in newProducts) {
-                                            if (p.productIdentifier) {
-                                                self.productsByIdentifier[p.productIdentifier] = p;
-                                            }
-                                        }
-                                    }
-                                    CALL_IF_SET_ON_MAIN_THREAD(completion, [products arrayByAddingObjectsFromArray:newProducts]);
-                                }];
+        [self.productsManager productsWithIdentifiers:missingProductIdentifiers
+                                           completion:^(NSSet<SKProduct *> * _Nonnull newProducts) {
```

Comment:
this one had mostly the same API as `requestFetcher`, except that it returned a set instead of an array